### PR TITLE
Update to morepork-only detection model

### PR DESCRIPTION
**Pull request by dsosnoski**

### CODE REVIEW

File: Melt/identify_species.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    # accumulate results from all models
+    activations_sum = np.zeros(len(samples))
+    model_paths = _model_paths(models)
+    for path in model_paths:
+        model = tf.keras.models.load_model(path)
+        activations = model.predict(samples).flatten()
+        activations_sum += activations
+
+    def entry(begin, end, species, activation):
```
Comment (gferraro) [inline Melt/identify_species.py]: THis should just be a standalone method

File: Melt/identify_species.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        entry['begin_s'] = begin * sample_slide_seconds
+        entry['end_s'] = (i - 1) * sample_slide_seconds + seconds_per_sample
+        entry['species'] = species
+        entry['liklihood'] = round(activation * 0.01, 2)
+        return entry
+
+    # generate labels from summed activations
+    labels = []
+    liklihoods = [round(v * 33.33333333) for v in activations_sum]
+    first = -1
```
Comment (gferraro) [inline Melt/identify_species.py]: maybe just make this None
Is `first` and `last` refering to the start and end of a morepork sample? A better name might be `sample_start, sample_end`

File: Melt/identify_species.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    pathlist = list()
+    for name in namelist:
+        namepath = os.path.join(basepath, name)
+        if os.path.isdir(namepath):
+            pathlist = pathlist + _model_paths(namepath)
+        elif namepath.endswith(model_file_name):
+            pathlist.append(basepath)
+    return pathlist
+
+def _find_likely_span(liklihoods, first, last):
```
Comment (gferraro) [inline Melt/identify_species.py]: not really sure what is going on here, but looks like it's always returning  `max(liklihoods[first:last + 1])`  so just calculate that once. is count  the number of samples?

It looks to me like all of these conditions could be condensed and written generallically using similar logic to line:69

File: Melt/identify_species.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    count = last - first
+    if count == 0:
+        end = first + seconds_per_sample
+        return liklihoods[first], first, end
+    elif count == 1:
+        liklihood = max(liklihoods[first], liklihoods[last])
+        end = first + seconds_per_sample
+        return liklihood, first + 1, end
+    elif count == 2:
+        max_liklihood = max(liklihoods[first:last + 1])
```
Comment (gferraro) [inline Melt/identify_species.py:71]: just use ` np.argmin` and `np.argmax` here and then compare the indexes

File: Melt/identify_species.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        entry = {}
+        entry['begin_s'] = begin * sample_slide_seconds
+        entry['end_s'] = (i - 1) * sample_slide_seconds + seconds_per_sample
+        entry['species'] = species
+        entry['liklihood'] = round(activation * 0.01, 2)
+        return entry
+
+    # generate labels from summed activations
+    labels = []
+    liklihoods = [round(v * 33.33333333) for v in activations_sum]
```
Comment (gferraro) [inline Melt/identify_species.py]: `v * 100 / 3.0` guessing this is because there is 3 models?

File: requirements.txt [hunk 6]
```diff
@@ -1,9 +1,9 @@
 h5py~=2.10.0
 Keras-Applications~=1.0.8
 Keras-Preprocessing~=1.1.0
-numpy~=1.17.3
+numpy~=1.19.2
 scipy~=1.3.1
-tensorflow==2.0.0
+tensorflow~=2.4.1
 Wave~=0.0.2
```
Comment (gferraro) [inline requirements.txt:9]: does it need librosa?

File: Melt/chain.py [hunk 7]
```diff
@@ -107,17 +37,7 @@ def main():
         ci = cacophony_index.calculate(argv[2])
         summary.update(ci)
     elif argv[1] == '-examine':
```
Comment (gferraro) [inline Melt/chain.py]: if we arenot going to have any other argument options lets just remove the need for -examine

File: Melt/identify_species.py (hunk 3 above)
Comment (gferraro) [reply Melt/identify_species.py]: Is this returning the span in seconds? to make this clearer, the parameters could be `sample_start`, `sample_end` and it returns `start_s`, `end_s` or something like that and a bit of a description about what this function is actually doing

**Review by gferraro — COMMENTED** [review]
Hey looks good, just a few things to look at.

File: Melt/identify_species.py (hunk 4 above)
Comment (dsosnoski) [reply Melt/identify_species.py:71]: I return the actual max_liklihood value, so I don't see that argmin/argmax would simplify this noticeably. It'd be a little simpler for min_liklihood, but I'd rather stay consistent.

File: Melt/identify_species.py (hunk 3 above)
Comment (dsosnoski) [reply Melt/identify_species.py]: rewrote this to document and clarify what's going on

File: Melt/identify_species.py (hunk 1 above)
Comment (dsosnoski) [reply Melt/identify_species.py]: okay

File: Melt/identify_species.py (hunk 2 above)
Comment (dsosnoski) [reply Melt/identify_species.py]: rewrote this to make it easier to understand

File: Melt/identify_species.py (hunk 5 above)
Comment (dsosnoski) [reply Melt/identify_species.py]: yes, made this dependency on the number of models explict

File: requirements.txt (hunk 6 above)
Comment (dsosnoski) [reply requirements.txt:9]: sure do. I thought there would be an error if I missed a dependency...

File: Melt/chain.py (hunk 7 above)
Comment (dsosnoski) [reply Melt/chain.py]: okay

**Review by gferraro — APPROVED** [review]
Cool be good to see how this goes

**Pull request merged**